### PR TITLE
Remote improvements

### PR DESCRIPTION
**Pull request by sixten-hilborn**
Includes PR #12 and parts of #11 (better control over remote repos).

### CODE REVIEW

**Comment by solvingj** [comment]
Sorry, perhaps the logic is hard to follow and I should have commented better.  I guess I wasn't sure I was done figuring it out yet.  

To be clear, the current logic is intended to be as follows: 
- If user has defined `CONAN_UPLOAD`: pass it
- Else: use the automatic deduction function
- If user has added `CONAN_REMOTES`: pass them directly
- Else: use the upload

Is this how you understood it to work? If so, can you give the "user story" of the case that is not possible with this approach?  Even with your great comments, I can't infer what we're missing with the current approach.

**Comment by sixten-hilborn** [comment]
The name `split_colon_env` sure is misleading since it creates a Python `list` from a comma-separated list, not a colon-separated list. I should clarify that in the comments.

I think we have the same understanding of the current logic, however I believe it's not as convenient as it could be. If I want to create a new package (following the guidelines) that has references to other bincrafters packages, I'd first create a repo under my user. Then I need to add the bincrafters/public-conan URL myself since it's not added by default (this is what I want to change with this PR).

I don't have any strong opinions regarding having `CONAN_REMOTES` override vs prepend the default remote list though.

**Comment by solvingj** [comment]
Ok, I see.  Driving home now, will think about it, sounds like you're probably right.

**Comment by sigmoidal** [comment]
@solvingj I just wanted to add that I introduced `split_colon_env` from the conan internals because this is the way conan itself parses the variables that recipe authors ought to specify in their recipes and thus format can be considered "conan-compliant".

the logic of the original PR in #11 regarding remotes was that conan public and bincrafters remotes are typically necessary in our own build.py across packages, since there are deps, as @sixten-hilborn mentions, so more often than not it will be necessary to provide again and again those same remotes. Instead, take user-supplied remotes (if provided) and prepend these predefined remotes, so that when users want to override them they can, without losing the ability to have a cascading remote search functionality.

The original explanation and first code is here: https://github.com/bincrafters/conan-glog/pull/1 for reference. This also had another advantage with split-builds on appveyor per build type, but it got lost along the way. 

I propose that @sixten-hilborn's PR is integrated, as it solves more than one problems. thanks @sixten-hilborn gj.
